coderd/debug.go

```diff
@@ -89,12 +94,36 @@ func (api *API) debugDeploymentHealth(rw http.ResponseWriter, r *http.Request) {
 		})
 		return
 	case res := <-resChan:
-		formatHealthcheck(ctx, rw, r, res.Val)
+		report := res.Val
+		if report == nil {
+			httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+				Message: "There was an unknown error completing the healthcheck.",
+				Detail:  "nil report from healthcheck result channel",
+			})
+			return
+		}
+		formatHealthcheck(ctx, rw, r, *report, dismissed...)
 		return
 	}
 }
 
-func formatHealthcheck(ctx context.Context, rw http.ResponseWriter, r *http.Request, hc *healthcheck.Report) {
+func formatHealthcheck(ctx context.Context, rw http.ResponseWriter, r *http.Request, hc healthcheck.Report, dismissed ...codersdk.HealthSection) {
+	// Mark any sections previously marked as dismissed.
+	for _, d := range dismissed {
+		switch d {
+		case codersdk.HealthSectionAccessURL:
+			hc.AccessURL.Dismissed = true
+		case codersdk.HealthSectionDERP:
+			hc.DERP.Dismissed = true
+		case codersdk.HealthSectionDatabase:
+			hc.Database.Dismissed = true
+		case codersdk.HealthSectionWebsocket:
+			hc.Websocket.Dismissed = true
+		case codersdk.HealthSectionWorkspaceProxy:
+			hc.WorkspaceProxy.Dismissed = true
+		}
+	}
+
 	format := r.URL.Query().Get("format")
 	switch format {
 	case "text":
```

Comment:
not part of this PR: we could update it to use severity + dismissed as well.

Reply:
Will address in a follow-up.